packages/components/src/List/Toolbar/SelectDisplayMode/SelectDisplayMode.component.js

```diff
@@ -1,55 +1,74 @@
 import React, { PropTypes } from 'react';
 import { Nav, NavDropdown, MenuItem } from 'react-bootstrap';
 import uuid from 'uuid';
+import _intersection from 'lodash/intersection';
 
 import Icon from '../../../Icon';
 
 function getIcon(selected) {
 	switch (selected) {
-	case 'table': return 'talend-table';
-	case 'large': return 'talend-expanded';
-	case 'tile': return 'talend-tiles';
-	default: return 'talend-table';
+	case 'table':
+		return 'talend-table';
+	case 'large':
+		return 'talend-expanded';
+	case 'tile':
+		return 'talend-tiles';
+	default:
+		return 'talend-table';
 	}
 }
 
 function getLabel(selected) {
 	switch (selected) {
-	case 'table': return 'Table';
-	case 'large': return 'Expanded';
-	case 'tile': return 'Tile';
-	default: return 'Table';
+	case 'table':
+		return 'Table';
+	case 'large':
+		return 'Expanded';
+	case 'tile':
+		return 'Tile';
+	default:
+		return 'Table';
 	}
 }
 
 const options = ['table', 'large', 'tile'];
 
-function SelectDisplayMode({ id, mode, onChange }) {
+function SelectDisplayMode({ id, mode, availableModes, onChange }) {
 	const selected = mode || 'table';
 	const displayIcon = (<Icon name={getIcon(selected)} />);
+
 	function onChangeMode(value, event) {
 		return onChange(event, value);
 	}
 
+	function getAvailablesOptions(modes) {
+		if (modes) {
+			return _intersection(modes, options);
```

Comment:
as we have lodash on dependencies, why not use it ?
i don't think too it's necessary here btw, just found it cleaner.